language-design.md

```diff
@@ -38,8 +57,8 @@ How do the themes of _Growing a Language_ relate to the lab we did this week?
 
 **Response**
 
-
-
+I think the methodology of adding functionality is probably the biggest similarity between the two. Steele introduces words only as needed, and some of the words he introduces (like "woman") are only introduced in order to introduce other words or ideas. He starts with an existing framework -- what he calls Basic English, and only expands on it based on the use cases that he needs. I think this is quite similar to what we did in the lab. We looked at the existing framework that we had, which is the sample solution to the CS5 sounds lab, and introduced features as we needed them. First we added a playSound function and a global variable storing the sound, which gave us the ability to delay the playing of the sound until all the transformations had been done. We only introduced features and functionality as needed, and didn't add things that would have been nice to have but weren't crucial (similar to the "-ly" which Steele decided not to include in the end).
+The other thing that I found similar to Steele's talk was the idea that a poorly implemented but intuitive language is better than a well-implemented but difficult to expand language. Steele talks about this with his discussion of APL, which was created by "a very smart man" [Steele, 1998], but didn't have staying power because adding to the language was extremely difficult. He compares this with Lisp and Java, which gave the power to the programmers and as such has been able to stay around until this day. I think this is similar to the lab in that we tried to make thing easy to understand and expand upon rather than making things perfect. For example, we made a saveSound function which allowed users to save sounds and name their files. While we could have been much more expansive in this process, or perhaps made the syntax for saving sounds require less keystrokes, we intended to make it as intuitive as possible for the user to simply save and name a sound.
 ---
```

Comment:
This "starting small and building up" is something I was trying to comment on as well when I answered this question, but you did a much better job of explaining the connection.

I actually thought that the lab and the reading were more different than they were similar.  Steele's biggest thing was about language growth, and we didn't really address that in our lab.  In fact, we were very focused on our end user base and on predicting exactly what they'd want with out DSL.